static/app/views/performance/browser/resources/resourceSummaryPage/resourceSummaryCharts.tsx

```diff
@@ -1,30 +1,86 @@
 import {SpanMetricsField} from 'sentry/views/starfish/types';
 import {getDurationChartTitle} from 'sentry/views/starfish/views/spans/types';
-import {Block} from 'sentry/views/starfish/views/spanSummaryPage/block';
+import {Block, BlockContainer} from 'sentry/views/starfish/views/spanSummaryPage/block';
+
+const {SPAN_SELF_TIME, HTTP_RESPONSE_CONTENT_LENGTH} = SpanMetricsField;
 
 function ResourceSummaryCharts(props: {groupId: string}) {
   const {data: spanMetricsSeriesData, isLoading: areSpanMetricsSeriesLoading} =
-    useSpanMetricsSeries(props.groupId, {}, [`avg(${SpanMetricsField.SPAN_SELF_TIME})`]);
+    useSpanMetricsSeries(props.groupId, {}, [
+      `avg(${SPAN_SELF_TIME})`,
+      `avg(${HTTP_RESPONSE_CONTENT_LENGTH})`,
+    ]);
 
   return (
-    <Block>
-      <ChartPanel title={getDurationChartTitle('http')}>
-        <Chart
-          height={160}
-          data={[spanMetricsSeriesData?.[`avg(${SpanMetricsField.SPAN_SELF_TIME})`]]}
-          loading={areSpanMetricsSeriesLoading}
-          utc={false}
-          chartColors={[AVG_COLOR]}
-          isLineChart
-          definedAxisTicks={4}
-        />
-      </ChartPanel>
-    </Block>
+    <BlockContainer>
+      <Block>
+        <ChartPanel title={getDurationChartTitle('http')}>
+          <Chart
+            height={160}
+            data={[spanMetricsSeriesData?.[`avg(${SPAN_SELF_TIME})`]]}
+            loading={areSpanMetricsSeriesLoading}
+            utc={false}
+            chartColors={[AVG_COLOR]}
+            isLineChart
+            definedAxisTicks={4}
+          />
+        </ChartPanel>
+      </Block>
+      <Block>
+        <ChartPanel title={t('Resource Size')}>
+          <Chart
+            height={160}
+            data={[spanMetricsSeriesData?.[`avg(${HTTP_RESPONSE_CONTENT_LENGTH})`]]}
+            loading={areSpanMetricsSeriesLoading}
+            utc={false}
+            chartColors={[AVG_COLOR]}
+            isLineChart
+            definedAxisTicks={4}
+            tooltipFormatterOptions={{
+              valueFormatter: bytes =>
+                getDynamicText({
+                  value: formatBytesBase2(bytes),
+                  fixed: 'xx KB',
+                }),
+            }}
+          />
+        </ChartPanel>
+      </Block>
+    </BlockContainer>
   );
 }
 
+/**
+ * Ensures a series has no zeros between two non-zero datapoints. This is useful in
+ * @param series the series to fill
+ * @returns a reference to the initial series filled
+ */
+export const fillSeries = (series: Series): Series => {
+  const data = series.data;
+
+  if (!data.length) {
+    return series;
+  }
+
+  let lastSeenValue = data[0].value;
+  for (const dataPoint of data) {
+    const value = dataPoint.value;
+    if (value !== lastSeenValue && value !== 0) {
+      lastSeenValue = value;
+    } else {
+      dataPoint.value = lastSeenValue;
+    }
+  }
+  series.data = series.data.filter(dataPoint => dataPoint.value > 0);
+  return series;
+};
```

Comment:
l: I know we're not using this right now, but for later - I prefer if we don't mutate the incoming argument if the function does a return. Immutable objects helps with tricky state issues sometimes.

Maybe something like this? (I wrote in GH code editor so please try before using).

```ts
const fillSeries = (series: Series): Series => {
  if (!series.data.length) {
    return series;
  }

  return {
    ...series,
    data: series.data.map((dataPoint, index) => {
      if (index === 0) {
        return dataPoint;
      }

      const prevValue = series.data[index - 1].value;
      const value = dataPoint.value === 0 ? prevValue : dataPoint.value;
      return { ...dataPoint, value };
    }),
  }
};
```

Reply:
Yeah I like that pattern and can update it. 

On a side note, JavaScript actually does this natively in some cases, array.sort will modify the array as well as return the array. Not saying it's fine because JS does it, more of a fun fact! Not a fan of how that's done.

Reply:
done!

Reply:
yeah `array.sort()` has this because of backwards compatibility reasons. This was the behavior with most array APIs back when they were released in JS 1.1, all the way back in 1996 (Netscape Navigator 3!)

It's weird because `array.reverse()` which was introduced at that time did not have that same return behaviour.

If you look at the [JS 1.1 spec](https://hepunx.rl.ac.uk/~adye/jsspec11/builtin.htm#67144), it actually doesn't explicitly state if the method should return something.

Reply:
That's pretty interesting, thanks for sharing!